sentry-android-core/src/test/java/io/sentry/android/core/ViewHierarchyEventProcessorTest.kt

```diff
@@ -315,6 +315,72 @@ class ViewHierarchyEventProcessorTest {
         assertFalse(fixture.options.sdkVersion!!.integrationSet.contains("ViewHierarchy"))
     }
 
+    @Test
+    fun `when view hierarchies are captured rapidly, capturing should be debounced`() {
+        val processor = fixture.getSut(true)
+        val event = SentryEvent().apply {
+            exceptions = listOf(SentryException())
+        }
+        val hint0 = Hint()
+        processor.process(event, hint0)
+        assertNotNull(hint0.viewHierarchy)
+
+        val hint1 = Hint()
+        processor.process(event, hint1)
+        assertNull(hint1.viewHierarchy)
+    }
+
+    @Test
+    fun `when view hierarchies are captured rapidly, debounced flag should be propagated`() {
+        val processor = fixture.getSut(true)
+
+        var debounceFlag = false
+        fixture.options.setBeforeViewHierarchyCaptureCallback { _, _, debounce ->
+            debounceFlag = debounce
+            true
+        }
+        val event = SentryEvent().apply {
+            exceptions = listOf(SentryException())
+        }
+        val hint0 = Hint()
+        processor.process(event, hint0)
+        assertFalse(debounceFlag)
+
+        val hint1 = Hint()
+        processor.process(event, hint1)
+        assertTrue(debounceFlag)
+    }
+
+    @Test
+    fun `when capture callback returns false, no view hierarchy should be captured`() {
+        fixture.options.setBeforeViewHierarchyCaptureCallback { _, _, _ ->
+            false
+        }
+        val (_, hint) = fixture.process(
+            true,
+            SentryEvent().apply {
+                exceptions = listOf(SentryException())
+            }
+        )
+
+        assertNull(hint.viewHierarchy)
+    }
+
+    @Test
+    fun `when capture callback returns true, a view hierarchy should be captured`() {
+        fixture.options.setBeforeViewHierarchyCaptureCallback { _, _, _ ->
+            true
+        }
+        val (_, hint) = fixture.process(
+            true,
+            SentryEvent().apply {
+                exceptions = listOf(SentryException())
+            }
+        )
+
+        assertNotNull(hint.viewHierarchy)
+    }
+
```

Comment:
could you add a last test `when capture callback returns true, debounce flag is overwritten`?
basically a test where the debounce flag is false and the callback returns true, so the view hierarchy should be attached anyway

Reply:
Oh yeah I missed that, thanks! This test case actually caught an edge case 🙇